### PR TITLE
[PA-997] Add support for NFS Backup Location Addition

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR makes changes to support NFS Backup Location addition. This will help improve the test coverage.

**Which issue(s) this PR fixes** (optional)
Closes #PA-997,PA-1158

**Special notes for your reviewer**:
~~As of the time of writing, deleting backups in the NFS Backup Location is not supported, making it impossible to run tests involving the deletion of created backups.~~

Please review the Jenkins build for the "BasicBackupCreation" test case using the link below

~~https://jenkins.pwx.dev.purestorage.com/job/Users/job/Sumit/job/Custom-Pipelines/job/px-backup-on-demand-system-test-byoc/1431/ (hard-coded parameters)~~

~~Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/203298~~

Thanks to the efforts of @ak-px, the "BasicBackupCreation" test case was successfully executed using a dedicated BYOC pipeline for NFS backup location. Please review the following Jenkins builds:

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/NFS/job/PX-Backup-System-Test-(BYOC)-NFS/7

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/208325

https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/NFS/job/PX-Backup-System-Test-(BYOC)-NFS/8 (FB NFS Endpoint)

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/208332

### CODE REVIEW

**Review by vpinisetti-px — CHANGES_REQUESTED** [review]

File: tests/common.go [hunk 1]
```diff
@@ -3716,6 +3720,87 @@ func CreateAzureBackupLocation(name string, uid string, cloudCred string, cloudC
...
+	if err != nil {
+		return fmt.Errorf("failed to wait for backup location addition. Error:[%v]", err)
+	}
+	return nil
+}
+
+// CreateNFSBackupLocation creates backuplocation for nfs
+func CreateNFSBackupLocation(name string, uid string, orgID string) error {
+	serverAddr := os.Getenv("NFS_SERVER_ADDR")
+	if serverAddr == "" {
```
[the remark this answers is not in the document]
Comment (kphalgun-px) [reply tests/common.go]: Done. Thanks!

File: tests/common.go [hunk 2]
```diff
@@ -3595,6 +3597,8 @@ func CreateCloudCredential(provider, credName string, uid, orgID string, ctx con
 				},
 			},
 		}
+	case drivers.ProviderNfs:
+		return nil
```
Comment (mkoppal-px) [inline tests/common.go:3601]: Add a log message before return stating `Creation of NFS backuplocation does not require Cloud Credential`

Comment (kphalgun-px) [reply tests/common.go:3602]: Done. Added a warning message saying cloud credential creation is not required for NFS provider.

File: tests/common.go [hunk 3]
```diff
@@ -3716,6 +3721,81 @@ func CreateAzureBackupLocation(name string, uid string, cloudCred string, cloudC
...
+		return err
+	}
+	_, err = backupDriver.CreateBackupLocation(ctx, bLocationCreateReq)
+	if err != nil {
+		return fmt.Errorf("failed to create backup location Error: %v", err)
+	}
+	err = WaitForBackupLocationAddition(ctx, name, uid, orgID, defaultTimeout, defaultRetryInterval)
+	if err != nil {
+		return err
+	}
```
Comment (vpinisetti-px) [inline tests/common.go:3795]: We need to handle this for negative scenario. Because this always checks for happy path.

**Review by vpinisetti-px — CHANGES_REQUESTED** [review]

File: tests/common.go (hunk 3 above)
Comment (kphalgun-px) [reply tests/common.go]: Done. Added a validate bool parameter to allow the test user to decide whether to validate the NFS backup location.

**Review by kshithijiyer-px — APPROVED** [review]
LGTM

**Review by vpinisetti-px — APPROVED** [review]
LGTM

**Review by sn-px — CHANGES_REQUESTED** [review]
Create and delete of mount point needs to be added as part of the before/after suite like how we do it for s3 bucket

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Review by sn-px — APPROVED** [review]
Approved .

**Comment by ak-px** [comment]
Please add the NFS env variable to deployments/deploy-ssh.sh .
for ref : https://github.com/portworx/torpedo/pull/1374/

**Comment by kphalgun-px** [comment]
> Please add the NFS env variable to deployments/deploy-ssh.sh . for ref : #1374

Done. Thank you.

**Review by ak-px — APPROVED** [review]
LGTM

File: tests/common.go [hunk 4]
```diff
@@ -3522,6 +3522,8 @@ func CreateBackupLocation(provider, name, uid, credName, credUID, bucketName, or
 		err = CreateS3BackupLocation(name, uid, credName, credUID, bucketName, orgID, encryptionKey)
 	case drivers.ProviderAzure:
 		err = CreateAzureBackupLocation(name, uid, credName, CloudCredUID, bucketName, orgID)
+	case drivers.ProviderNfs:
+		err = CreateNFSBackupLocation(name, uid, orgID, true)
```
Comment (mkoppal-px) [inline tests/common.go:3526]: We need to take in encryption key

Comment (mkoppal-px) [reply tests/common.go:3526]: As suggested by @snigdha-px

File: tests/common.go [hunk 5]
```diff
@@ -3716,6 +3721,83 @@ func CreateAzureBackupLocation(name string, uid string, cloudCred string, cloudC
...
+	}
+	_, err := task.DoRetryWithTimeout(f, timeout, timeBeforeRetry)
+	if err != nil {
+		return fmt.Errorf("failed to wait for backup location addition. Error:[%v]", err)
+	}
+	return nil
+}
+
+// CreateNFSBackupLocation creates backup location for nfs
+func CreateNFSBackupLocation(name string, uid string, orgID string, validate bool) error {
```
[the remark this answers is not in the document]
Comment (kphalgun-px) [reply tests/common.go]: I will try to resolve this in another PR.

**Pull request merged**